upcloud/network.go

```diff
@@ -88,15 +88,27 @@ func (s *Interface) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+type DHCPRoutesConfig struct {
+	EffectiveRoutesAutoPopulation EffectiveRoutesAutoPopulation `json:"effective_routes_auto_population"`
+}
+
+type EffectiveRoutesAutoPopulation struct {
+	Enabled             Boolean  `json:"enabled"`
+	ExcludeBySource     []string `json:"exclude_by_source,omitempty"`
+	FilterByDestination []string `json:"filter_by_destination,omitempty"`
+	FilterByRouteType   []string `json:"filter_by_route_type,omitempty"`
+}
+
 // IPNetwork represents an IP network in a response.
 type IPNetwork struct {
-	Address          string   `json:"address,omitempty"`
-	DHCP             Boolean  `json:"dhcp"`
-	DHCPDefaultRoute Boolean  `json:"dhcp_default_route"`
-	DHCPDns          []string `json:"dhcp_dns,omitempty"`
-	DHCPRoutes       []string `json:"dhcp_routes,omitempty"`
-	Family           string   `json:"family,omitempty"`
-	Gateway          string   `json:"gateway,omitempty"`
+	Address          string           `json:"address,omitempty"`
+	DHCP             Boolean          `json:"dhcp"`
+	DHCPDefaultRoute Boolean          `json:"dhcp_default_route"`
+	DHCPDns          []string         `json:"dhcp_dns,omitempty"`
+	DHCPRoutes       []string         `json:"dhcp_routes,omitempty"`
+	Family           string           `json:"family,omitempty"`
+	Gateway          string           `json:"gateway,omitempty"`
+	DHCPRoutesConfig DHCPRoutesConfig `json:"dhcp_routes_configuration,omitempty"`
```

Comment:
Could also use the full _configuration_ word in the naming to be more in sync with the API, but this works as well 🤔 For examples:

```suggestion
	DHCPRoutesConfiguration DHCPRoutesConfiguration `json:"dhcp_routes_configuration,omitempty"`
```